assignments/lambda-classes.js

```diff
@@ -1 +1,143 @@
 // CODE here for your Lambda Classes
+ //class-Person, base-class
+ class Person {
+     constructor(object) {
+         this.name = object.name;
+         this.age = object.age;
+         this.location = object.location;
+         this.gender = object.gender;
+     }
+     speak() {
+         return `Hello my name is ${this.name}, I am from ${this.location}.`;
+     }
+ }
+
+
+ //Instructors - extensions of Person ie Person-parent and instructor-child
+class Instructor extends Person 
+{
+    constructor(instructor)
+    {
+        super(instructor);
+        this.speciality = instructor.speciality;
+        this.favLanguage = instructor.favLanguage;
+        this.catchPhrase = instructor.catchPhrase;
+    }
+    demo(subject) 
+    {
+      return (`Today we are learning about ${subject}.`);
+    }
+    grade(student, subject) 
+    {
+    // return (`${student.name} receives a perfect score on ${subject}.`);
+    return (`${student} receives a perfect score on ${subject}.`);  
+    }
+}
+
+
+ //Student - extensions of Person ie Person-parent and student-child
+class Student extends Person 
+{
+    constructor(student)
+    {
+        super(student);
+        this.previousBackground  = student.previousBackground ;
+        this.className  = student.className ;
+        this.favSubjects = student.favSubjects;
+       
+    }
+    listsSubject()
+    {
+        for (let i = 0; i < this.favSubjects.length; i++)
+        {
+            return (`${this.name} favourite subjects are ${this.favSubjects[i]}`);
```

Comment:
Seems like `this.favSubjects[i]` is only returning the first subject. Look into the spread operator `...`. It would be used like so: `[...this.favSubjects]`